src/esssans/loki/io.py

```diff
@@ -79,121 +82,169 @@ def _merge_events(a, b):
 
 def _merge_runs(
     data_groups: sciline.Series[
-        Filename[ScatteringRunType], LoadedSingleFileContents[ScatteringRunType]
+        Filename[ScatteringRunType], LoadedSingleFileDetector[ScatteringRunType]
```

Comment:
Below, the name is saying this can be either detector or monitor data, but here we are saying it only applies to detector data?
I know it's just an internal function whose types are not checked when building the pipeline, but I still got a bit confused when reading.